fvm/transactionInvocator.go

```diff
@@ -158,9 +163,12 @@ func (i *TransactionInvocator) Process(
 		txError = fmt.Errorf("transaction invocation failed: %w", err)
 	}
 
-	// check the storage limits
-	if ctx.LimitAccountStorage && txError == nil {
-		txError = NewTransactionStorageLimiter().Process(vm, ctx, proc, sth, programs)
+	if txError == nil {
+		txError = i.checkAccountStorageLimit(vm, ctx, proc, sth, programs)
+	}
+
+	if txError == nil {
+		txError = i.deductTransactionFees(env, proc)
```

Comment:
I guess we need to change this logic, when it fails we still have to charge fees right?

Reply:
I was going to leave that change for another PR. I don't want to include that change in this spork.